src/core/server/saved_objects/routes/import.ts

```diff
@@ -75,13 +76,29 @@ export const registerImportRoute = (router: IRouter, config: SavedObjectConfig)
       },
     },
     router.handleLegacyErrors(async (context, req, res) => {
-      const { overwrite, createNewCopies } = req.query;
+      const { overwrite, createNewCopies, dataSourceId } = req.query;
       const file = req.body.file as FileStream;
       const fileExtension = extname(file.hapi.filename).toLowerCase();
       if (fileExtension !== '.ndjson') {
         return res.badRequest({ body: `Invalid file extension ${fileExtension}` });
       }
 
+      // get datasource from saved object service
+      // dataSource is '' when there is no dataSource pass in the url
+      const dataSource = dataSourceId
+        ? await context.core.savedObjects.client
+            .get('data-source', dataSourceId)
+            .then((response) => {
+              const attributes: any = response?.attributes || {};
+              return {
+                id: response.id,
+                title: attributes.title,
+              };
+            })
```

Comment:
I am having trouble understanding how `await <>.get()` is returning a promise which is then used with a `then`.

Also, do we have a test where the repsonse is missing `attributes.title`? (to see how the `then` works).